client/src/app/pages/external/jira/useUpdatingTrackerIds.ts

```diff
@@ -0,0 +1,33 @@
+import { useState, useEffect } from "react";
+
+export type UpdatableId = string | number | null;
+
+const useUpdatingTrackerIds = (delay: number = 5000) => {
+  const [updatingTrackerIds, setUpdatingTrackerIds] = useState<
+    Set<UpdatableId>
+  >(new Set());
+
+  useEffect(() => {
+    let timerId: number | null = null;
+
+    if (updatingTrackerIds.size > 0) {
+      timerId = window.setTimeout(() => {
+        setUpdatingTrackerIds(new Set());
+      }, delay);
+    }
+
+    return () => {
+      if (timerId !== null) {
+        window.clearTimeout(timerId);
+      }
+    };
```

Comment:
Does this work if you updated 2 trackers within the 10 second interval?

The `addUpdatingTrackerId` function looks right, but the removal of the ids seems off since I don't see something like `new Set(prevSet.remove(expiredId))`.

Reply:
It is actually working because on refetch this hook is remounted / cleaned up. Only works because we are routinely polling for trackers.

Reply:
Ah I see what you mean. The entire set is wiped when one timeout expires. Fixing.